File: java_solutions/1720.decode-xored-array.java
/*
 * @lc app=leetcode id=1720 lang=java
 *
 * [1720] Decode XORed Array
 */

class Solution {
    public int[] decode(int[] encoded, int first) {
        return Stream.iterate(new int[]{first, 0}, a -> new int[]{a[0] ^ encoded[a[1]], a[1] + 1})
            .limit(encoded.length + 1)
            .mapToInt(a -> a[0])
            .toArray();
    }
}

File: kotlin_solutions_1/1720.decode-xored-array.kt
/*
 * @lc app=leetcode id=1720 lang=kotlin
 *
 * [1720] Decode XORed Array
 */

class Solution { fun decode(encoded: IntArray, first: Int): IntArray = IntArray(encoded.size + 1).also { it[0] = first }.also { arr -> encoded.forEachIndexed { i, v -> arr[i + 1] = arr[i] xor v } } }